storage/metric_directory.go

```diff
@@ -39,24 +41,43 @@ func (md *MetricDirectory) generateParentDirectories() []MetricDirectory {
 	return parents
 }
 
+// UpdateDirectories is a recursive function that will test missing directories
+// and insert them
 func (md *MetricDirectory) UpdateDirectories(storageConnector BgMetadataStorageConnector) error {
 	entry, err := storageConnector.SelectDirectory(md.name)
 	if err != nil {
-		if entry == "" {
-			parentDirectories := md.generateParentDirectories()
-			for _, pd := range parentDirectories {
-				err = storageConnector.InsertDirectory(pd)
-				if err != nil {
-					fmt.Println(err.Error())
-				}
+		if entry != "" {
+			p := NewMetricDirectory(md.parent)
+			p.UpdateDirectories(storageConnector)
+			err = storageConnector.InsertDirectory(md)
+			if err != nil {
+				return fmt.Errorf("cannot insert metric directory: %s", err.Error())
 			}
-		} else {
-			return fmt.Errorf("cannot check metric directory in cassandra: %s", err.Error())
 		}
 	}
-	err = storageConnector.InsertDirectory(*md)
-	if err != nil {
-		fmt.Println(err.Error())
-	}
 	return nil
 }
+
+// getUUID returns a hash of the name
+func (md *MetricDirectory) getUUID() string {
+	namespace, _ := uuid.Parse("00000000-1111-2222-3333-444444444444")
+	sha := uuid.NewSHA1(namespace, []byte(md.name))
+	return sha.String()
+}
+
+// ToESDocument returns a json string that represents the document
+func (md *MetricDirectory) ToESDocument() string {
+	var components = strings.Split(md.name, ".")
+	var b strings.Builder
+	b.WriteString(fmt.Sprintf(`{"name":"%s",`, md.name))
```

Comment:
Why using a string builder over json serialization ? If good rational ( performance ? ) I think it would be a good idea to add a comment to explain it.